### PR TITLE
Adapt to OpenFisca-Core 17.0.0

### PR DESCRIPTION
**Pull request by michelbl**
Connected to https://github.com/openfisca/openfisca-core/issues/523
* Do not use `compose_reforms()`
* Accept Openfisca-Core 17.0.0 in requirements

### CODE REVIEW

**Comment by michelbl** [comment]
La API web est cassée par le passage en YAML, notamment https://github.com/openfisca/openfisca-web-api/blob/master/openfisca_web_api/controllers/swagger.py

Par exemple, le type des valeurs (float, int, bool, string) n'est pas explicite dans YAML alors que swagger l'expose.

A quel point l'interface swagger est critique et doit rester stable ?

**Comment by Anna-Livia** [comment]
Concernant les types de valeur, il y a un "unit" dans le Yaml --> est-ce que le format des valeur demandé par le swagger pourrait être inféré à partir de ça ?

**Comment by MattiSG** [comment]
From https://github.com/openfisca/openfisca-core/pull/552#issuecomment-322783180:

> > It seems to be the Web API, which cannot serve the type and format properties anymore.
> 
> If regaining the type property is easy thanks to Python introspection, let's bring it back and not fill in > the format optional property at all (or expose the unit contents).
> 
> We just need to obtain the list of currently exposed type parameters. This can be done by inspecting the swagger endpoint.

**Comment by MattiSG** [comment]
> The API needs to keep on outputing the type of the variables

Why? Who is consuming it? For what purpose?

**Comment by michelbl** [comment]
Pour connaitre les types et formats utilisés : 
```
jq '..|.format?' swagger | sort | uniq -c
jq '..|.type?' swagger | sort | uniq -c
```

Je ne connais pas du tout ce code et je pense que m'y plonger dedans seul serait une grosse perte de temps, sans garantie de résultats acceptables à la fin. D'après l'historique, c'est @fpagnoux qui connait le mieux ce repo. Je propose donc de pairer avec lui ou le laisser seul adapter à la nouvelle représentation des paramètres.

**Comment by MattiSG** [comment]
Based on data from https://github.com/openfisca/openfisca-core/issues/523#issuecomment-313714903:

- Expose `unit` in the `type` property (both are freehand, optional fields).
- Infer `format` from Python type introspection. Only `percent` will be lost.

Beware, the `unit` variable was [already used](https://github.com/openfisca/openfisca-core/issues/523#issuecomment-313660324) to represent the XML `type` attribute, so it might already be called that way even if no code change was applied.

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,9 @@ addons:
   ssh_known_hosts: api.openfisca.fr
 install:
   - pip install --upgrade pip wheel  # pip >= 8.0 needed to be compatible with "manylinux" wheels, used by numpy >= 1.11
-  - pip install "OpenFisca-France==18.5.3"  # Needed only for tests.
+  - pip install --editable git+https://github.com/openfisca/openfisca-core.git@params-yaml#egg=OpenFisca-Core
```
Comment (michelbl) [inline .travis.yml]: Commit is called "[Read before merge] Travis configuration", not explicit enought.

**Comment by sandcha** [comment]
Using YAML parameters doesn't work with `openfisca_parsers.input_variables_extractor.get_input_variables_and_parameters(column)`. 

As `openfisca_parsers` module is deprecated and the result of `get_input_variables_and_parameters(column)` mainly used by the deprecated route `/graph`, we removed the `openfisca_parsers` dependency and the `/graph` route.

**Pull request merged**